modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c

```diff
@@ -18,66 +18,56 @@
 #include <kernel.h>
 #include <logging/log.h>
 #include <stdbool.h>
+#include <stdint.h>
+#include <zephyr.h>
 
 #define LOG_LEVEL LOG_LEVEL_INFO
 #define LOG_MODULE_NAME nrf_802154_temperature
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
-/** @brief Value of the last temperature measurement in degree Celsius. */
-static int8_t temperature;
+/** @brief Structure describing the temperature measurement module. */
+static struct {
+	const struct device *dev;	/**< Temperature device pointer. */
+	int8_t value;			/**< Last temperature measurement in degree Celsius. */
 
-void nrf_802154_temperature_init(void)
-{
-	/* Intentionally empty. */
-}
+	struct k_work_delayable work;
+} temperature;
 
-void nrf_802154_temperature_deinit(void)
+static void work_handler(struct k_work *work)
 {
-	/* Intentionally empty. */
-}
+	struct sensor_value temperature_value;
 
-int8_t nrf_802154_temperature_get(void)
-{
-	return temperature;
-}
+	sensor_sample_fetch(temperature.dev);
+	sensor_channel_get(temperature.dev, SENSOR_CHAN_DIE_TEMP, &temperature_value);
 
-/** @brief Thread handler for temperature update. */
-static void temperature_update_thread(void *p1, void *p2, void *p3)
-{
-	ARG_UNUSED(p1);
-	ARG_UNUSED(p2);
-	ARG_UNUSED(p3);
+	if (temperature.value != temperature_value.val1) {
+		temperature.value = temperature_value.val1;
 
-	static struct sensor_value  temperature_value;
-	static const struct device *temperature_dev;
+		nrf_802154_temperature_changed();
+	}
 
-	temperature_dev = DEVICE_DT_GET(DT_NODELABEL(temp));
+	k_work_reschedule(&temperature.work, K_MSEC(CONFIG_NRF_802154_TEMPERATURE_UPDATE_PERIOD))
+}
 
-	if (!device_is_ready(temperature_dev)) {
+void nrf_802154_temperature_init(void)
+{
+	temperature.dev = DEVICE_DT_GET(DT_NODELABEL(temp));
+
+	if (!device_is_ready(temperature.dev)) {
```

Comment:
asserts are turned off by default
if the code won't work when the device is not ready the error handler should emit an error and return. Assert can be left there to simplify debugging.

If we expect device is always ready (i.e. this situation cannot ever occur) there should be no error code, simply assert in form
```
__ASSERT_NO_MSG(device_is_ready(temperature.dev));
```

Reply:
We do not expect die temperature sensor would malfunction, hence only an assert is left.